adaptdl/adaptdl/torch/adascale.py

```diff
@@ -266,13 +276,19 @@ def _final_callback(self):
                                  for g in group] for group in grads]
 
         if count > 1:
-            grad_sqr = (count * total_sqr - local_sqr) / (count - 1)
-            grad_var = (local_sqr - total_sqr) * scale / (count - 1)
-            self.raw_grad_sqr = grad_sqr
-            self.raw_grad_var = grad_var
-            theta = self._smoothing ** scale
-            self._update_avg('sqr_avg', grad_sqr, theta)
-            self._update_avg('var_avg', grad_var, theta)
+            # Note: mixed precision can result in nan/inf gradients,
+            # which propogate into our norm and variance estimates.
+            # Mixed precision autoscaling skips the skip where
+            # there are nan/inf, so we also skip the update here
+            if (np.all(np.isfinite(total_sqr))
+                    and np.all(np.isfinite(local_sqr))):
```

Comment:
This check might need to be a bit earlier, since we'd probably want to skip lines 250-276 as well (and definitely lines 260-262 and 274).

Reply:
Pretty sure 
```
 if any(c != c or c in (float('inf'), -float('inf')) for c in check):
        print("AdaScale detected invalid gradient! Skipping step.")
        return
```
covers that as well. I think I can remove this final check.

Reply:
Sure, let's remove this then.